### PR TITLE
Update for release v1.24.0

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR updates docs, release notes and Dockerfiles for the upcoming ISPC release v1.24.0.

### CODE REVIEW

File: docker/v1.24.0/ubuntu18.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#  SPDX-License-Identifier: BSD-3-Clause
+
+ARG LLVM_VERSION=17.0
+
+FROM ubuntu:18.04 AS llvm_build_step1
+LABEL maintainer="Dmitry Babokin <dmitry.y.babokin@intel.com>"
+SHELL ["/bin/bash", "-c"]
+
+ARG REPO=ispc/ispc
+ARG SHA=main
```
Comment (aneshlya) [inline docker/v1.24.0/ubuntu18.04/Dockerfile:13]: I noticed that starting v1.22.0 we forgot updating SHA in Dockerfiles. Could you update it here and for Dockerfiles for previous releases?

**Review by dbabokin — APPROVED** [review]
LGTM

File: docker/v1.24.0/ubuntu18.04/Dockerfile (hunk 1 above)
Comment (nurmukhametov) [reply docker/v1.24.0/ubuntu18.04/Dockerfile]: Done

File: docs/ispc.rst [hunk 2]
```diff
@@ -595,6 +596,35 @@ A couple of improvements have been made to variables initialization:
...
 type.
 
+Updating ISPC Programs For Changes In ISPC 1.24.0
+-------------------------------------------------
+
+This release extends the standard library with new functions performing dot
+product operations. These functions utilize specific hardware instructions from
+AVX-VNNI and AVX512-VNNI. The ISPC targets that support native VNNI
+instructions are ``avx2vnni-i32x*``, ``avx512icl-*`` and ``avx512spr-*``. The
+first targets (``avx2vnni-*`` and ``avx512icl-*``) were introduced in this
```
[the remark this answers is not in the document]
Comment (nurmukhametov) [reply docs/ispc.rst]: Fixed

File: docs/ispc.rst [hunk 3]
```diff
@@ -595,6 +596,35 @@ A couple of improvements have been made to variables initialization:
...
+Now, uniform integers and enums can be used as non-type template parameters.
+Please refer to `Function Templates`_ for more details.
+
+The release contains the following changes that may affect compatibility with
+older versions:
+
+* ``--pic`` command line flag now corresponds to the ``-fpic`` flag of Clang
+  and GCC, whereas the newly introduced ``--PIC`` corresponds to ``-fPIC``. In
+  some cases, to preserve previous behavior, users may need to switch to
+  ``--PIC``.
```
[the remark this answers is not in the document]
Comment (nurmukhametov) [reply docs/ispc.rst]: Right, I tried to make it clear

[the remark this answers is not in the document]
Comment (dbabokin) [reply docs/ispc.rst]: I think it's good enough now.

**Pull request merged**